Add PO ExtraLineItems functionality to API

miggland opened the pull request:
I have copied the relevant functions for PO Line items so that they work for extra line items as well.

These were previously missing from the python API

SchrodingersGat commented:
@miggland thanks for this addition. It would be great if you can also add some unit tests for new new class functionality. The existing unit tests should be a pretty good demonstration of how to set that up

miggland commented:
OK, I have fixed the style issue and added unit tests according to the template

miggland commented:
I realised that the delete method was not working - because of a wrong url, caused by an extra / in the URL. This was copied from the line item, so I changed it there also. I also added tests for deleting line items, as well as extra line items

SchrodingersGat commented:
@miggland a couple of things to address here

### Style Fixes

A simple change needed here, just read through the suggestions in the "style checks" test

### Unit Tests

It appears that the CI unit testing script was not allowing enough time for the InvenTree server to spin up. This has been fixed in #103 . Please merge master into your branch, and re-push, and the tests should run properly then.

SchrodingersGat commented:
> I have tried getting the tests to run locally 

Did you run the `invoke update-image` command before running tests? That should get it all setup and ready to run

SchrodingersGat commented:
Well, at least the unit tests are running now!
